Feature/35 logout from session

ifirmawan opened the pull request:
## TODO/DONE

- [x] Remove sessions from db and state
- [x] Unit tests

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204955479635210

ifirmawan committed:
[#35] Create new component: LogoutButton

ifirmawan committed:
[#23] Add new prop: editable

ifirmawan committed:
[#23] Add conditional: editable

ifirmawan committed:
[#35] Add Logout button on SettingsPage

ifirmawan committed:
[#35] Clear state & session on successfull logout

ifirmawan committed:
[#35] Create unit test: LogoutButton

ifirmawan committed:
[#23] Update Settings test to replace server editable

ifirmawan committed:
[#35] Update snapshot: Settings to include logout

ifirmawan committed:
Merge branch 'main' into feature/35-logout-from-session

ifirmawan committed:
[#23] Replace all use React directly in SettingsForm

dedenbangkit commented on app/src/pages/Settings/Form.js:
@ifirmawan please change the name to SettingsForm instead Form.js

ifirmawan replied:
Okay mas, on it

dedenbangkit commented on app/src/components/LogoutButton.js:
```diff
@@ -0,0 +1,70 @@
+  const handleYesPress = () => {
+    const table = 'sessions';
+    const clearQuery = query.clear(table);
+    console.log('clear', clearQuery);
```
remove console log

dedenbangkit commented on app/src/components/LogoutButton.js:
```diff
@@ -0,0 +1,70 @@
+
+  const handleError = () => {
+    if (Platform.OS === 'android') {
+      ToastAndroid.show('Logout Failed: Please try again later', ToastAndroid.LONG);
```
why do we even need this?

dedenbangkit commented on app/src/components/LogoutButton.js:
```diff
@@ -0,0 +1,70 @@
+
+  const handleYesPress = () => {
+    const table = 'sessions';
+    const clearQuery = query.clear(table);
```
Please create truncate query instead of just clearing the session.

ifirmawan replied:
Hi mas @dedenbangkit actually clear is truncate query
https://github.com/akvo/nmis-mobile/blob/main/app/src/database/query.js#L25-L27

dedenbangkit replied:
Sorry, I mean the truncate should support multiple tables

ifirmawan committed:
[#23] Remove Settings/Form

ifirmawan committed:
[#23] Rename Form to SettingsForm

ifirmawan committed:
[#35] Remove console log & toast failed

ifirmawan committed:
[#23] Remove Form.test.js

ifirmawan committed:
[#23] Rename Form.test to SettingsForm.test

ifirmawan committed:
[#15] Handle multiple transactions

This is because a single transaction cannot execute string that containing multiple queries.

ifirmawan committed:
[#15] Refactoring clear to support multiple tables

ifirmawan committed:
[#35] Truncate sessions & users table

ifirmawan committed:
[#15] Update truncate test to handle multi tables

ifirmawan committed:
[#27] Add AuthState to store username

ifirmawan committed:
[#35] Set id null and empty string for password

ifirmawan committed:
[#27] Revert AddUserPage without authstate

ifirmawan committed:
[#23] Fix invalid username in users table and remove unnecessary code

ifirmawan committed:
[#35] Truncate all tables

ifirmawan committed:
[#35] Add rows count test in Logout
